internal/controllers/provisioningrequest_clusterconfig.go

```diff
@@ -155,6 +151,8 @@ func (t *provisioningRequestReconcilerTask) updateConfigurationAppliedStatus(
 			"The Cluster is not yet ready",
 		)
 		if utils.IsClusterProvisionCompleted(t.object) &&
+			(!utils.IsClusterUpgradeInitiated(t.object) ||
```

Comment:
Looks like
`!utils.IsClusterUpgradeInitiated(t.object) ||
				utils.IsClusterUpgradeCompleted(t.object)`
could be moved in its own function since it's been used 3 times, something like `IsClusterUpgradeInProgress`.
It can be done in another PR, for sure. I think it would greatly improve the code's readability.